packages/core/src/store/types.ts

```diff
@@ -7,3 +7,23 @@ export interface BucketConnect {
 export interface TableConnect {
   region: string
 }
+
+// Store records
+
+export interface DealStoreRecord {
+  // PieceCid of an Aggregate `bagy...aggregate`
+  piece: string
+  // address of the Filecoin storage provider storing deal
+  provider: string
+  // deal identifier
+  dealId: number
+  // epoch of deal expiration
+  expirationEpoch: number
+  // source of the deal information
+  source: string
+  // Date when deal was added as ISO string
+  insertedAt: string
```

Comment:
Might be nice to use native types like `Link`/`Date` where possible. This is something I've done in billing, the encode for stores converts to a `Record<string, string|number>` and the decode converts the record to the rich type.

e.g. 

https://github.com/web3-storage/w3infra/blob/8ef9d894593d6cbb6acad0e151103f63f550f06a/billing/lib/api.ts#L6-L23

I also have this handy type to automatically inter a record that is going to be stored in DynamoDB:

https://github.com/web3-storage/w3infra/blob/8ef9d894593d6cbb6acad0e151103f63f550f06a/billing/types.ts#L1-L6

...and so your encode function returns/decode function accepts:

```ts
InferStoreRecord<DealStoreRecord>
```

Reply:
oh cool thanks! This is quite nice, adopting!